src/components/ComputeModal.js

```diff
@@ -623,7 +623,8 @@ export const ComputeModalBase = ({ onDismiss, onSuccess, runtimes, persistentDis
         const gpuConfig = runtimeConfig?.gpuConfig
         const newSparkMode = Utils.switchCase(runtimeConfig?.cloudService,
           [cloudServices.DATAPROC, () => runtimeConfig.numberOfWorkers === 0 ? 'master' : 'cluster'],
-          [cloudServices.GCE, () => false]
+          [cloudServices.GCE, () => false],
+          [undefined, () => false] // for when there's no existing runtime
```

Comment:
I preferred `undefined` over `Utils.DEFAULT` for clarity.

Reply:
There is a subtle difference between `undefined` and `Utils.default`.

If `cloudService` is equivalent to `undefined` then `newSparkMode` will be set to false. However, if `cloudService` is something unexpected there will be no default case, resulting in `newSparkMode` getting set to `undefined`

Leaving `undefined` in may be ok, but do you may also want an explicit default case. If undefined is essentially the default, I would stick with `DEFAULT`.